graphql_compiler/compiler/emit_sql.py

```diff
@@ -440,18 +451,28 @@ def _construct_fold_subquery(self, subquery_from_clause):
         else:
             raise NotImplementedError(
                 "Fold only supported for MSSQL and "
-                "PostgreSQL, dialect was set to {}".format(self._dialect.name)
+                "PostgreSQL, dialect was set to {self._dialect.name}."
             )
 
-    def _get_array_agg_column(self, intermediate_fold_output_name, fold_output_field):
+    def _get_array_agg_column(
+        self, intermediate_fold_output_name: str, fold_output_field: str
+    ) -> Label:
         """Select an array_agg of the fold output field, labeled as requested."""
+        if self._output_vertex_alias is None:
+            raise AssertionError(
+                "Attempted to perform array aggregation while the "
+                "_output_vertex_alias was set to None."
```

Comment:
In situations like this, I'd suggest adding as much of the current state into the error message as possible, to ease reproducing and debugging the issue. That would include things like function arguments and relevant `self` attributes. Future-us will be grateful to you for saving them a painful and frustrating debugging step :)